modules/cassandrabackend/cassandrabackend.cc

```diff
@@ -0,0 +1,324 @@
+		cassandradbmanager::getInstance();
+		L << Logger::Info << " Connection to cassandra cluster successful " << endl;
+
+	 } catch (const ArgException &A) {
+		L << Logger::Error << "[CassandraBackend]" << " Fatal argument error: "<< A.reason << endl;
+		throw;
+	 } catch (...) {
+		throw;
+	 }
+
+  }
+
+  bool list(const string &target, int domain_id, bool include_disabled=false)
+  {
+    return false; // we don't support AXFR
+  }
+
+  bool getSOA(const string &name, SOAData &soadata, DNSPacket *p=0) {
+	 domain = name;
+	 L << Logger::Info << "[CassandraBackend] Recieved getSOA " <<domain<< " "<< endl;
+	 if(all_authoritative_server == 1) {
+		 L << Logger::Info << "[CassandraBackend] All authoritative server enabled, sending default SOA"<<endl;
+		 populatedefaults(soadata);
+		 return true;
+	 }
+	 fetchdata();
+	 for (int index = 0; index < totalSize; ++index) {
+		 backendrecord backendRecord = backendRecords[index];
+		 if(backendRecord.getType() == QType::SOA) {
+			 L << Logger::Info << "[CassandraBackend] SOA record found"<< endl;
+			 soadata.db = this;
+			 //Data format :: domain_id nameserver	hostname    serial refresh retry expiry default_ttl
+			 std::string data = backendRecord.getRecord();
+			 boost::char_separator<char> delimiter("#");
+			 boost::tokenizer<boost::char_separator<char> > tokens(data, delimiter);
+			 int rec_index  = 0;
+			 BOOST_FOREACH(std::string const& token, tokens)
+			 {
+				 switch(rec_index) {
+				 	 case 0:
+				 		 soadata.domain_id = (uint32_t)atoi(token.c_str());
+				 		 break;
+				 	case 1:
+						 soadata.nameserver = token.c_str();
+						 break;
+				 	case 2:
+						 soadata.hostmaster = token.c_str();
+						 break;
+				 	case 3:
+						 soadata.serial = (uint32_t)atoi(token.c_str());
+						 break;
+				 	case 4:
+						 soadata.refresh = (uint32_t)atoi(token.c_str());
+						 break;
+				 	case 5:
+						 soadata.retry = (uint32_t)atoi(token.c_str());
+						 break;
+				 	case 6:
+						 soadata.expire = (uint32_t)atoi(token.c_str());
+						 break;
+				 	case 7:
+						 soadata.default_ttl = (uint32_t)atoi(token.c_str());
+						 break;
+				 	default:
+				 		L << Logger::Info << "[CassandraBackend]Extra params while parsing" << endl;
+				 }
+				 soadata.ttl = backendRecord.getTtl();
+				 rec_index++;
+			 }
+			 L << Logger::Info << "[CassandraBackend] SOA record true"<< endl;
+			 L << Logger::Info << "[CassandraBackend] SOA serial "<<soadata.serial<<" refresh "<<soadata.refresh<<" retry "<<soadata.retry
+					 <<" expire "<<soadata.expire<<" default_ttl "<<soadata.default_ttl<<" domain_id "<<soadata.domain_id<<" ttl "<<soadata.ttl
+					 <<" nameserver "<<soadata.nameserver<<" hostmaster "<<soadata.hostmaster<<endl;
+			 clear();
+			 return true;
+		 }
+	}
+	 L << Logger::Info << "[CassandraBackend] SOA record false"<< endl;
+	 clear();
+	 return false;
+
+  }
+
+  void populatedefaults(SOAData &soadata) {
```

Comment:
I was thinking of having a configurable SOA, if the user chooses cassandra-all-authoritative-server mode. WIP